src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -0,0 +1,957 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// A source method that can have attributes, including a member method, accessor, or local function.
+    /// </summary>
+    internal abstract class SourceMethodSymbolWithAttributes : SourceMethodSymbol, IAttributeTargetSymbol
+    {
+        private CustomAttributesBag<CSharpAttributeData> _lazyCustomAttributesBag;
+        private CustomAttributesBag<CSharpAttributeData> _lazyReturnTypeCustomAttributesBag;
+
+        // some symbols may not have a syntax (e.g. lambdas, synthesized event accessors)
+        protected readonly SyntaxReference syntaxReferenceOpt;
+        protected SourceMethodSymbolWithAttributes(SyntaxReference syntaxReferenceOpt)
+        {
+            this.syntaxReferenceOpt = syntaxReferenceOpt;
+        }
+
+#nullable enable
+        internal CSharpSyntaxNode? GetMethodBodyNode()
```

Comment:
>GetMethodBodyNode [](start = 35, length = 17)

As you mentioned, this also returns a constructor initializer. Perhaps we should reflect this in the name to avoid surprises for potential consumers

Reply:
Yeah, `GetBinderContextNode` was just taken from the local `contextNode` that the method was factored out from. Maybe there's a name that describes it accurately and also will make sense to compiler-internal consumers.